### PR TITLE
add ver.ParseVersion

### PR DESCRIPTION
**Pull request by ryanking**

### CODE REVIEW

File: ver/version.go [hunk 1]
```diff
@@ -43,3 +44,21 @@ func versionString(version, sha string, release, dirty bool) string {
...
+		dirty = true
+		v = strings.TrimSuffix(v, ".dirty")
+	}
+	if strings.Contains(v, "-") {
+		tmp := strings.Split(v, "-")
+		v = tmp[0]
+		sha = tmp[1]
+	}
+
+	semVersion, _ := semver.Parse(v)
```
Comment (edulop91) [inline ver/version.go]: why ignore error?

File: ver/version.go [hunk 2]
```diff
@@ -43,3 +44,21 @@ func versionString(version, sha string, release, dirty bool) string {
 	}
 	return fmt.Sprintf("%s+%s+dirty", version, sha)
 }
+
+func ParseVersion(version string) (semver.Version, string, bool) {
+	var dirty bool
+	var sha string
+	v := version
```
Comment (edulop91) [inline ver/version.go]: probably don't need this reassignment

File: ver/version.go (hunk 1 above)
Comment (ryanking) [reply ver/version.go]: good question, this was copied as-is from fogg

Comment (ryanking) [reply ver/version.go]: ah it appears we currently don't do any error handling

**Pull request merged**